filter.js

```diff
@@ -199,3 +181,28 @@ Filter.prototype.processString =
       'When subclassing cauliflower-filter you must implement the ' +
       '`processString()` method.');
 };
+
+function memoize(func, cacheName) {
+  if (typeof func !== 'function') throw new TypeError('Expected a function');
+  function memoized(self, key) {
+    var cache = self[cacheName] || (self[cacheName] = Object.create(null));
+    var entry = cache[key];
+    if (entry !== void 0) return entry;
+
+    var args = [];
+    for (var i = 1, ii = arguments.length; i < ii; ++i) args.push(arguments[i]);
+    return cache[key] = func.apply(self, args);
```

Comment:
your correct, as long as getDestFilePath doesn't for some unknown reason return `undefined` we are ok. The only scenario is if someone abuses .extensions, which seems extremely unlikely. If someone does this, they deserve the slightly worse caching :P